simulator/__tests__/singleton.spec.ts

```diff
@@ -0,0 +1,110 @@
+import { Runtime, Account } from "..";
+import { main } from "asbuild";
+import { promisify } from "util";
+import { join } from "path";
+
+let runtime: Runtime;
+let singleton: Account, alice: Account;
+
+function getErrorMsg(res: any) {
+  try {
+    return res.err["FunctionCallError"]["HostError"]["GuestPanic"].panic_msg;
+  } catch (e) {
+    throw new Error(JSON.stringify(res.err, null, 2));
+  }
+}
+
+async function compile(contract: string): Promise<void> {
+  function asb(succ: any, fail: any) {
+    main([join(__dirname,"../assembly/__tests__", contract + ".ts",), "--target", "debug", "--wat"], {
+    },
+    (err) => {
+      if (err) {
+        throw(err);
+        return -1;
+      } else {
+        succ();
+        return 1;
+      }
+    })
+  }
+  return new Promise(asb);
+}
+
+describe("Complier fails", () => {
+  it("shouldn't allow methods with the same name as init function", async () => {
+    try {
+      await compile("singleton-fail")
+      expect(true).toBe(false);
+    } catch (e) {
+      expect(e.message).toContain(`Method "new" already used cannot export constructor using the same name.`);
+    }
+  })
+})
```

Comment:
This seems like a good place to test that you can use the `@nearBindgen` decorator on a class with no `constructor`